csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/NugetPackageRestorer.cs

```diff
@@ -589,7 +593,26 @@ private static async Task ExecuteGetRequest(string address, HttpClient httpClien
         private bool IsFeedReachable(string feed, int timeoutMilliSeconds, int tryCount, bool allowExceptions = true)
         {
             logger.LogInfo($"Checking if Nuget feed '{feed}' is reachable...");
-            using HttpClient client = new();
+
+            // Configure the HttpClient to be aware of the Dependabot Proxy, if used.
+            HttpClientHandler httpClientHandler = new();
+            if (this.dependabotProxy.IsConfigured)
+            {
+                httpClientHandler.Proxy = new WebProxy(this.dependabotProxy.Address);
+
+                if (!String.IsNullOrEmpty(this.dependabotProxy.CertificatePath))
+                {
+                    X509Certificate2 proxyCert = new X509Certificate2(this.dependabotProxy.CertificatePath);
```

Comment:
This is because I haven't manage to fully manage to get the extractor to target .NET9 yet. I expect that when https://github.com/github/codeql/pull/18091 is merged then this can be fixed.